subsys/net/lib/dns/resolve.c

```diff
@@ -755,14 +813,73 @@ int dns_resolve_cancel(struct dns_resolve_context *ctx, u16_t dns_id)
 	return 0;
 }
 
+int dns_resolve_cancel_with_name(struct dns_resolve_context *ctx,
+				 u16_t dns_id,
+				 const char *query_name,
+				 enum dns_query_type query_type)
+{
+	u16_t query_hash = 0;
+
+	if (query_name) {
+		struct net_buf *buf;
+		u16_t len;
+		int ret;
+
+		/* Use net_buf as a temporary buffer to store the packed
+		 * DNS name.
+		 */
+		buf = net_buf_alloc(&dns_msg_pool, ctx->buf_timeout);
+		if (!buf) {
+			return -ENOMEM;
+		}
+
+		/* If the query string + \0 + query type (A or AAAA)
+		 * does not fit the tmp buf, then bail out
+		 */
+		len = strlen(query_name);
+		if ((len + 1 + 2) > buf->size) {
+			net_buf_unref(buf);
+			return -ENOMEM;
+		}
+
+		ret = dns_msg_pack_qname(&len, buf->data, buf->size,
+					 query_name);
+		if (ret >= 0) {
+			len = strlen(buf->data);
```

Comment:
dns_msg_pack_qname() already sets the len, so no need to recompute it.

That removes again a cpu greedy strlen()